src/app/server/JointFabricDatastore.cpp

```diff
@@ -122,16 +122,166 @@ CHIP_ERROR JointFabricDatastore::RemoveNode(NodeId nodeId)
+    {
+        auto it =
+            std::find_if(mEndpointEntries.begin(), mEndpointEntries.end(),
+                         [&endpoint](const Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type & entry) {
+                             return entry.endpointID == endpoint.id;
+                         });
+        if (it == mEndpointEntries.end())
+        {
+            Clusters::JointFabricDatastore::Structs::DatastoreEndpointEntryStruct::Type newEntry;
+            newEntry.endpointID = endpoint.id;
+            newEntry.nodeID     = nodeId;
+            mEndpointEntries.push_back(newEntry);
+        }
+    }
+
+    // Remove EndpointEntries that are not in the endpointsList
+    for (auto it = mEndpointEntries.begin(); it != mEndpointEntries.end();)
+    {
+        if (it->nodeID == nodeId)
+        {
+            bool found = false;
+
+            for (const auto & endpoint : endpointsList)
+            {
+                if (it->endpointID == endpoint.id)
+                {
+                    found = true;
+                    break;
+                }
+            }
+
+            if (!found)
+            {
+                // If the endpoint is not in the endpointsList, we remove it
+                it = mEndpointEntries.erase(it);
+                continue;
+            }
+        }
+
+        ++it;
+    }
+
+    for (auto it = mEndpointGroupIDEntries.begin(); it != mEndpointGroupIDEntries.end();)
+    {
+        if (it->nodeID == nodeId)
+        {
+            if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kPending)
+            {
+                // If the node is pending, we need to update the status entry to Committed
+                it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+            }
+            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kDeletePending)
+            {
+                // If the node is delete pending, we need to remove the entry
+                it = mEndpointGroupIDEntries.erase(it);
+                continue;
+            }
+            else if (it->statusEntry.state == Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitFailed)
+            {
+                CHIP_ERROR failureCode(it->statusEntry.failureCode);
+
+                if (failureCode == CHIP_ERROR_CONSTRAINT || failureCode == CHIP_ERROR_RESOURCE_EXHAUSTED)
+                {
+                    // remove entry from the list
+                    it = mEndpointGroupIDEntries.erase(it);
+                    continue;
+                }
+                else
+                {
+                    // If the node is pending, we need to update the status entry to Committed
+                    it->statusEntry.state = Clusters::JointFabricDatastore::DatastoreStateEnum::kCommitted;
+                }
+            }
+        }
+
+        ++it;
+    }
+
```

Comment:
```suggestion
    // TODO: fetch binding list from node
```

Reply:
Isn't the datastore already supposed to reflect this information? Or should this sync (if needed) be handled outside of the RefreshNode command?